cicerone/app/src/main/java/com/teampower/cicerone/GeofencingController.kt

```diff
@@ -0,0 +1,71 @@
+package com.teampower.cicerone
+
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.util.Log
+import com.google.android.gms.location.Geofence
+import com.google.android.gms.location.GeofencingClient
+import com.google.android.gms.location.GeofencingRequest
+import com.google.android.gms.location.LocationServices
+
+class GeofencingController() {
+    private lateinit var geofencingClient: GeofencingClient
+    private var geofenceList = ArrayList<Geofence>()
+
+    fun startGeofencing(context: Context) {
+        val geofencePendingIntent: PendingIntent by lazy {
+            val intent = Intent(context, GeofenceBroadcastReceiver::class.java)
+            // We use FLAG_UPDATE_CURRENT so that we get the same pending intent back
+            // when calling addGeofences() and removeGeofences().
+            PendingIntent.getBroadcast(context, 0, intent, PendingIntent.FLAG_UPDATE_CURRENT)
+        }
+
+        geofencingClient = LocationServices.getGeofencingClient(context)
+        initializeGeofenceList()
+        geofencingClient?.addGeofences(getGeofencingRequest(), geofencePendingIntent)?.run {
+            addOnSuccessListener {
+                Log.v(TAG_GEO, "Geofences added")
+            }
+            addOnFailureListener{
+                Log.v(TAG_GEO, "Failed to add geofences")
+            }
+        }
+    }
+
+    private fun initializeGeofenceList() {
```

Comment:
Is this the function we need to change when using actual POI's later? This is where the list of geofences we want to set up for the user is generated right?

Reply:
My plan was for us to use createGeofence for the POIs, and them add them by using addGeofences. Then restart by calling startGeofencing. But that might not be optimal.